src/engine/pieces/piece.ts

```diff
@@ -13,6 +14,35 @@ export default class Piece {
         throw new Error('This method must be implemented, and return a list of available moves');
     }
 
+    public getLateralSquares(board: Board) {
```

Comment:
I would make the argument that this is not a method on a piece, but rather a method on the board.

The main thing that suggests this is that `GameSettings.BOARD_SIZE` is quite explicitly a property of the board (even if we've pulled it out as configuration here), but here it's leaked out of the `Board` class.

We're also only really acting on a square here; we're not using any functionality of the piece except to use the board to retrieve a square.

It would probably be clearer if this method was of the board, and you passed in a square.

Much of the same applies to the diagonal code below.